securitycenter/docs/snippets_findings.py

```diff
@@ -0,0 +1,603 @@
+    client = securitycenter.SecurityCenterClient()
+
+    # Field mask to only update the display name.
+    field_mask = field_mask_pb2.FieldMask(paths=["display_name"])
+
+    # source_name is the resource path for a source that has been
+    # created previously (you can use list_sources to find a specific one).
+    # Its format is:
+    # source_name = "organizations/{organization_id}/sources/{source_id}"
+    # e.g.:
+    # source_name = "organizations/111122222444/sources/1234"
+    updated = client.update_source(
+        {"name": source_name, "display_name": "Updated Display Name"},
+        update_mask=field_mask,
+    )
+    print("Updated Source: {}".format(updated))
+    # [END update_source]
+    assert updated.display_name == "Updated Display Name"
+
+
+def test_add_user_to_source(source_name):
+    """Gives a user findingsEditor permission to the source."""
+    user_email = "csccclienttest@gmail.com"
+    # [START update_source_iam]
+    from google.cloud import securitycenter as securitycenter
+    from google.iam.v1 import policy_pb2
+
+    client = securitycenter.SecurityCenterClient()
+
+    # source_name is the resource path for a source that has been
+    # created previously (you can use list_sources to find a specific one).
+    # Its format is:
+    # source_name = "organizations/{organization_id}/sources/{source_id}"
+    # e.g.:
+    # source_name = "organizations/111122222444/sources/1234"
+    # Get the old policy so we can do an incremental update.
+    old_policy = client.get_iam_policy(source_name)
+    print("Old Policy: {}".format(old_policy))
+
+    # Setup a new IAM binding.
+    binding = policy_pb2.Binding()
+    binding.role = "roles/securitycenter.findingsEditor"
+    # user_email is an e-mail address known to Cloud IAM (e.g. a gmail address).
+    # user_mail = user@somedomain.com
+    binding.members.append("user:{}".format(user_email))
+
+    # Setting the e-tag avoids over-write existing policy
+    updated = client.set_iam_policy(
+        source_name, {"etag": old_policy.etag, "bindings": [binding]}
+    )
+
+    print("Updated Policy: {}".format(updated))
+
+    # [END update_source_iam]
+    assert any(
+        member == "user:csccclienttest@gmail.com"
+        for member in chain.from_iterable(
+            binding.members for binding in updated.bindings
+        )
+    )
+
+
+def test_list_source(organization_id):
+    """Lists finding sources."""
+    i = -1
+    # [START list_sources]
+    from google.cloud import securitycenter as securitycenter
+
+    # Create a new client.
+    client = securitycenter.SecurityCenterClient()
+    # organization_id is the numeric ID of the organization. e.g.:
+    # organization_id = "111122222444"
+    org_name = "organizations/{org_id}".format(org_id=organization_id)
+
+    # Call the API and print out each existing source.
+    for i, source in enumerate(client.list_sources(org_name)):
+        print(i, source)
+    # [END list_sources]
+    assert i >= 0
```

Comment:
This is an unusual way to check for a non-empty list. Maybe save the returned list first, then check its length?

Reply:
list_sources is an iterator not a list.  On a prior PR i provided guidance not to copy to a list if it isn't recommended for clients to do (which in general I don't think it is since there can be many results).